R/runner.R

```diff
@@ -4,9 +4,15 @@
 ##'
 ##' @title Orderly runner
 ##' @param path Path to use
-##' @param allow_ref Allow git to change branches/ref for run
+##'
+##' @param allow_ref Allow git to change branches/ref for run.  If not
+##'   given, then we will look to see if the orderly configuration
+##'   disallows branch changes (based on the
+##'   \code{ORDERLY_API_SERVER_IDENTITY} environment variable and the
+##'   \code{master_only} setting of the relevant server block.
+##'
 ##' @export
-orderly_runner <- function(path, allow_ref = TRUE) {
+orderly_runner <- function(path, allow_ref = NULL) {
```

Comment:
if a user creates an `orderly_runner` with `allow_ref = TRUE`, we ignore the `master_only` in the `orderly_config`.

So a user could still run a report from a non-master branch on a remote wirh `master_only`, (although it would be much more difficult to do accidentally).

is this behaviour we want or should the `master_only` option override the `allow_ref` argument?

Reply:
You are correct. I'm not sure what the Right Thing to do here is but we will deprecate the argument soon I think